### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by acaltas** [henrywhitaker3/Speedtest-Tracker#558]
Hello

i am new on Docker and i tried to install our Speedtest-Tracker on my Synology NAS with the Docker APP

i got this error and the Docker Kills itself

2021-04-08 20:08:54 | stdout | [s6-finish] sending all processes the KILL signal and exiting.
-- | -- | --
2021-04-08 20:08:51 | stdout | [s6-finish] sending all processes the TERM signal.
2021-04-08 20:08:51 | stdout | [s6-finish] waiting for services.
2021-04-08 20:08:51 | stdout | [cont-finish.d] done.
2021-04-08 20:08:51 | stdout | [cont-finish.d] executing container finish scripts...
2021-04-08 20:08:51 | stdout | [cont-init.d] 50-speedtest: exited 1.
**2021-04-08 20:08:51 | stdout | chmod: cannot access '/config/www/app/Bin/speedtest': No such file or directory**
2021-04-08 20:08:48 | stdout | Clearing old jobs from queue
2021-04-08 20:08:48 | stdout | [32mDisabling authentication[39m
2021-04-08 20:08:47 | stdout | AUTH variable not set. Disabling authentication
2021-04-08 20:08:47 | stdout | Base path is unset
2021-04-08 20:08:47 | stdout | Telegram chat id and bot token unset
2021-04-08 20:08:47 | stdout | Slack webhook is unset

**Comment by Rubeast** [comment]
Hi! I had the same problem!
For some reason it would not download the needed application.
What I did to resolve this:

- Stop the container
- Download this file: https://bintray.com/ookla/download/download_file?file_path=ookla-speedtest-1.0.0-x86_64-linux.tgz
- extract it to \www\app\Bin
- Start container

**Comment by acaltas** [comment]
thx that fix it

**Comment by madtempest** [comment]
@Rubeast that "fixes" it but there's clearly an issue in code where the path in the container isn't pointing at the correct URL - thanks for the temp fix though!

**Comment by Rubeast** [comment]
Yeah I wasn't sure it was a problem with my config. So I looked for a fix for myself. I agree. Should be fixed in the code. At least he knows where he has to look.

**Comment by AndyYangUK** [comment]
> Hi! I had the same problem!
> For some reason it would not download the needed application.
> What I did to resolve this:
> 
> * Download this file: https://bintray.com/ookla/download/download_file?file_path=ookla-speedtest-1.0.0-x86_64-linux.tgz
> * extract it to \www\app\Bin
> * Start or restart the container

That file doesn't exist anymore :(

**Comment by Rubeast** [comment]
Link is also broken on their site: https://www.speedtest.net/apps/cli

**Comment by AndyYangUK** [comment]
Seems like it's an issue we need to wait for Ookla / speedtest.net team to sort out, unless someone has a mirror of that file for now?

**Comment by thierrybla** [comment]
Here is a quick fix to make it run atleast, however the speedtests itself are not working for me.

1. Download the file: https://install.speedtest.net/app/cli/ookla-speedtest-1.0.0-x86_64-linux.tgz
2. Make a folder called speedtest in \www\app\Bin and move the file from above into it
3. Restart the container

It should show you the panel now but my speedtest are not working.

**Comment by Rubeast** [comment]
> Here is a quick fix to make it run atleast, however the speedtests itself are not working for me.
> 
> 1. Download the file: https://install.speedtest.net/app/cli/ookla-speedtest-1.0.0-x86_64-linux.tgz
> 2. Make a folder called speedtest in \www\app\Bin and move the file from above into it
> 3. Restart the container
> 
> It should show you the panel now but my speedtest are not working.

I already told how to fix it above. You shouldn't make a folder called speedtest. Just extract the content of the archive under Bin.

**Comment by nickm324** [comment]
> > Here is a quick fix to make it run atleast, however the speedtests itself are not working for me.
> > 
> > 1. Download the file: https://install.speedtest.net/app/cli/ookla-speedtest-1.0.0-x86_64-linux.tgz
> > 2. Make a folder called speedtest in \www\app\Bin and move the file from above into it
> > 3. Restart the container
> > 
> > It should show you the panel now but my speedtest are not working.
> 
> I already told how to fix it above. You shouldn't make a folder called speedtest. Just extract the content of the archive under Bin.

Is there an arm version of this?

**Comment by Rubeast** [comment]
Yeah. Just download the Linux version from the official site https://www.speedtest.net/apps/cli

**Comment by nickm324** [comment]
> Yeah. Just download the Linux version from the official site https://www.speedtest.net/apps/cli

Yea I did that and after it runs it has an empty bin folder and many errors about not being able to download different things.

**Comment by madtempest** [comment]
Is there any reason this is using a binary and not the python script? I found that the python script is way more accurate than this binary.

**Comment by henrywhitaker3** [comment]
It originally used the python script, but it's not accurate at all for fast connections (in a container or not). For anyone who can replicate this, can you paste the output of the docker logs please?

**Comment by OmidTahouri** [comment]
@henrywhitaker3 
I first installed speedtest-tracker on 2021-04-18, which is after when this issue was created. Everything was working fine until today, when I decided (instant regret 🙂) to reinstall from scratch and haven't been able to get things running again.

`https://bintray.com/ookla/download/download_file?file_path=ookla-speedtest-1.0.0-x86_64-linux.tgz` 302s to `https://dl.bintray.com/ookla/download/ookla-speedtest-1.0.0-x86_64-linux.tgz?expiry=000&signature=AAA` which is returning a 403

It's strange that this was working fine for me after this GitHub issue was originally created. Perhaps some inconsistency or rate limiting on the bintray.com side (?).

A more complete log output is below (there's not much to go by). Note that this will indefinitely loop, starting with the linuxserver ASCII art and GID/UID values which I've omitted from this output. 

```
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-config: executing...
[cont-init.d] 20-config: exited 0.
[cont-init.d] 30-keygen: executing...
using keys found in /config/keys
[cont-init.d] 30-keygen: exited 0.
[cont-init.d] 40-config: executing...
Starting 2019/12/30, GeoIP2 databases require personal license key to download. Please manually download/update the GeoIP2 db and save as /config/geoip2db/GeoLite2-City.mmdb
[cont-init.d] 40-config: exited 0.
[cont-init.d] 50-speedtest: executing...
Ookla GDPR and EULA accepted. Downloading Speedtest CLI.
Connecting to bintray.com (75.126.118.184:443)
Connecting to dl.bintray.com (3.124.41.190:443)
wget: server returned error: HTTP/1.1 403 Forbidden

tar: can't open 'speedtest.tgz': No such file or directory
cp: cannot stat 'speedtest': No such file or directory
s6-applyuidgid: fatal: unable to exec /site/app/Bin/speedtest: No such file or directory
Copying latest site files to config
Database file exists
Env file exists
Running database migrations
Nothing to migrate.
App key exists
JWT secret exists
Slack webhook set, updating db

Not enough arguments (missing: "webhook").

Telegram chat id and bot token unset
Base path is unset
AUTH variable set, but not to 'true'. Disabling authentication
Disabling authentication
Clearing old jobs from queue
chmod: cannot access '/config/www/app/Bin/speedtest': No such file or directory
[cont-init.d] 50-speedtest: exited 1.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing...
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing...
usermod: no changes
```

**Comment by henrywhitaker3** [comment]
Yeah you're getting a 403 when trying to download the binary. Are you able to download it and place it in the volume manually?

**Comment by OmidTahouri** [comment]
That was my next step. Will report back...

**Comment by OmidTahouri** [comment]
Is there an alternative download link I can use?

**Comment by OmidTahouri** [comment]
I can confirm that the app can bootstrap successfully if the speedtest CLI binary is made available at `/config/www/app/Bin/speedtest` prior to starting.

Downloaded from https://www.speedtest.net/apps/cli > Linux > x86_64

**Comment by sOblivionsCall** [comment]
@henrywhitaker3 have you considered packaging the binary as part of the container build itself? this will allow you to have strict control of what binary version is packaged and will prevent this issue. That said i know that also causes more work for you, having to build a new container every time a new binary is released, just a thought.

**Comment by henrywhitaker3** [comment]
@sOblivionsCall only reason I'm not doing that is due to their license (as far as I can remember you're not allowed to distribute it)

**Comment by sOblivionsCall** [comment]
> @sOblivionsCall only reason I'm not doing that is due to their license (as far as I can remember you're not allowed to distribute it)

I'm surprised you were even able to find their license... so far i haven't found anything of the sort. Did you contact their support team to get it?

**Comment by OmidTahouri** [comment]
> > @sOblivionsCall only reason I'm not doing that is due to their license (as far as I can remember you're not allowed to distribute it)
> 
> I'm surprised you were even able to find their license... so far i haven't found anything of the sort. Did you contact their support team to get it?

https://www.speedtest.net/about/eula 🙂

> 2. Restrictions On Use. You shall not: [...] (e) rent, lease, lend, sell, sublicense, assign, **distribute**, publish, transfer, or otherwise make available the Software

**Comment by GGORG0** [comment]
Had the same problem in #735, after downloading the file manually still doesnt work

**Comment by gabriel-vanca** [comment]
This is still not working after all this time. Still getting the same error.

**Comment by acaltas** [comment]
on Synology Docker to fix it is easy
important Mount Path is set to a Folder what you can access by Windows Explorer 
![grafik](https://user-images.githubusercontent.com/82180703/139540480-9f8e6b35-e057-473b-85e2-ada0a0ab803b.png)

then download the File here and extra it into docker\speedtest\www\app\Bin
[speedtest.zip](https://github.com/henrywhitaker3/Speedtest-Tracker/files/7447054/speedtest.zip)
re

**Comment by elmocito** [comment]
> on Synology Docker to fix it is easy important Mount Path is set to a Folder what you can access by Windows Explorer ![grafik](https://user-images.githubusercontent.com/82180703/139540480-9f8e6b35-e057-473b-85e2-ada0a0ab803b.png)
> 
> then download the File here and extra it into docker\speedtest\www\app\Bin [speedtest.zip](https://github.com/henrywhitaker3/Speedtest-Tracker/files/7447054/speedtest.zip) re

that helped me as well. thanks @acaltas !!

**Comment by gabriel-vanca** [comment]
Hi. Was this ever resolved?
